pipenv/project.py

```diff
@@ -578,59 +573,37 @@ def clear_pipfile_cache(self):
         _pipfile_cache.clear()
 
     @staticmethod
-    def dump_dict(dictionary, write_to, inline=False):
-        """
-        Perform a nested recursive translation of a dictionary structure to a toml object.
-
-        :param dictionary: A base dictionary to translate
-        :param write_to: The root node which will be mutated by the operation
-        :param inline: Whether to create inline tables for dictionaries, defaults to False
-        :return: A new toml hierarchical document
-        """
-
-
-        def gen_table(inline=False):
-            if inline:
-                return tomlkit.inline_table()
-            return tomlkit.table()
+    def _is_tomlkit_parsed_result(parsed):
+        """Check by duck typing of tomlkit.api.Container"""
+        return hasattr(parsed, "_body")
 
-        for key, value in dictionary.items():
-            if isinstance(value, dict):
-                table = gen_table(inline=inline)
-                for sub_key, sub_value in value.items():
-                    if isinstance(sub_value, dict):
-                        table[sub_key] = Project.dump_dict(
-                            sub_value, gen_table(inline), inline=inline
-                        )
-                    else:
-                        table[sub_key] = sub_value
-                write_to[key] = table
-            else:
-                write_to[key] = Project.dump_dict(value, gen_table(inline), inline=inline)
+    @staticmethod
+    def convert_outline_table(parsed):
+        """Converts all outline to inline tables"""
+        if Project._is_tomlkit_parsed_result(parsed):
+            empty_inline_table = tomlkit.inline_table
         else:
-            write_to[key] = value
-        return write_to
+            empty_inline_table = toml.TomlDecoder().get_empty_inline_table
+        for section in ("packages", "dev-packages"):
+            table_data = parsed.get(section, {}).copy()
+            for package, value in table_data.items():
+                if hasattr(value, "keys"):
+                    table = empty_inline_table()
+                    table.update(value)
+                    table_data[package] = table
+            parsed[section] = table_data
+        return parsed
```

Comment:
I would prefer this (and `_is_tomlkit_parsed_result`) to be global functions, instead of under `Project` They are not really related to the project construct, but can be used in any general TOML structures.